wgsl/index.bs

```diff
@@ -11859,19 +11867,49 @@ See [[#builtin-inputs-outputs]] for how to specify that a pipeline input or outp
       <td>vertex
       <td>output
       <td>vec4&lt;f32&gt;
-      <td style="width:50%">Output position of the current vertex, using homogeneous coordinates.
-      After homogeneous normalization (where each of the *x*, *y*, and *z* components
-      are divided by the *w* component), the position is in the WebGPU normalized device
-      coordinate space.
-      See [[WebGPU#coordinate-systems]].
+      <td style="width:50%">Output position of the current vertex.
+
+      An output value (*x*,*y*,*z*,*w*)
+      [=behavioral requirement|will=] map to [=clip position=] (*x*/*w*, *y*/*w*, *z*/*w*) in
+      WebGPU [=normalized device coordinates=].
+
+      See [[WebGPU#coordinate-systems]] and [[WebGPU#primitive-clipping]].
 
   <tr>
       <td>fragment
       <td>input
       <td>vec4&lt;f32&gt;
-      <td style="width:50%">Framebuffer position of the current fragment in [=framebuffer=] space.
-      (The *x*, *y*, and *z* components have already been scaled such that *w* is now 1.)
-      See [[WebGPU#coordinate-systems]].
+      <td style="width:50%">
+      <div algorithm="fragment position calculation">
+      The |x|, |y|, and |z| components are the interpolated position of the current fragment,
+      in viewport coordinates. In the following, |vp| is the rasterization [=viewport=] state,
+      and |rp| is the [=RasterizationPoint=].
+      *  The |x| and |y| components are the interpolated position of the current fragment in
+          the [=framebuffer=].
+          It is |rp|.[=fragmentdestination-position|position=]
+
+          The framebuffer is a two-dimensional grid of pixels with the top-left at (0.0,0.0)
+          and the bottom right at (*vp.width*, *vp.height*).
+          Each pixel has an extent of 1.0 unit in each of the x and y dimensions,
+          and pixel centers are at (0.5,0.5) offset from integer coordinates.
+
+      * The |z| component is the interpolated depth of the current fragment,
+          scaled to the viewport depth range:
+
+           |z| = |vp|.minDepth - |rp|.[=rasterizationpoint-depth|depth=]
+               &times; (|vp|.maxDepth - |vp|.minDepth)
+
+          By default, |z| is clipped to the interval [|vp|.minDepth, |vp|.maxDepth].
+          It is *not* clipped when the primitive rasterization
+          state {{GPUPrimitiveState/unclippedDepth}} paramteter is true.
```

Comment:
Bit of confusion here between clipping and clamping. Attempted a rewrite.
```suggestion
          By default, geometry is clipped to the [0, 1] depth range of [=normalized device coordinates=].
          When the primitive rasterization state {{GPUPrimitiveState/unclippedDepth}} parameter is
          true, depth clipping does not occur, and |z| can be outside [|vp|.minDepth, |vp|.maxDepth].
          Depth *clamping* does not affect |z|, because it occurs after the fragment shader.
```

Reply:
Yeah, I botched that. Thank you again.

Its |rp|.depth.

This is given clearly in the API side's description of `process fragment`.  https://gpuweb.github.io/gpuweb/#abstract-opdef-process-fragment

Reply:
I think it's best to not mention clamping at all because that's way after the fragment shader.

Reply:
Rewritten.

Reply:
Personally I like mentioning things when there's potential for confusion, even if they're not supposed to be relevant. But not a strong preference here.

Reply:
Especially since you've removed mention of clipping.